### PR TITLE
Address benchmark inconsistencies in Annoy tutorial  #1105

### PR DESCRIPTION
**Pull request by droudy**
Issue #1105 

Uses average query time of 1000 random queries as opposed to only a single query. Includes a "dry run" before running queries. Also fixes a discrepancy where a comment says that the vector for "army" is being retrieved when the word is actually "science".  Benchmarks were ran on a 2.4GHz 4 core i7 processor.

### CODE REVIEW

File: docs/notebooks/annoytutorial.ipynb [hunk 1]
```diff
@@ -145,22 +141,34 @@
...
-      "('in', 0.9971976953092963)\n",
-      "('an', 0.9971216244157404)\n",
-      "('which', 0.9964874279685318)\n",
-      "('has', 0.996418114984408)\n",
-      "CPU times: user 4 ms, sys: 0 ns, total: 4 ms\n",
-      "Wall time: 3.22 ms\n"
+      "Gensim: 0.007451029\n",
+      "Annoy: 0.002149934\n",
+      "\n",
+      "Annoy is 3.46570127269 times faster on average over 1000 random queries\n"
```
Comment (piskvorky) [inline docs/notebooks/annoytutorial.ipynb]: The focus and emphasis on such a level of precision is misleading (and unnecessary).

Also, please mention the other factors that affect this number, like index size etc. So people don't go away thinking "annoy is ~3.5x faster than gensim", whereas in reality this is anything between 1x-infinity.

File: docs/notebooks/annoytutorial.ipynb [hunk 2]
```diff
@@ -269,16 +277,16 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "('orbits', 0.6323515474796295)\n",
-      "('changes,', 0.6309329569339752)\n",
-      "('dr.', 0.6305853128433228)\n",
-      "('terrorism,', 0.6300898194313049)\n",
-      "('creditors', 0.6264415979385376)\n"
+      "('signature', 0.5921074748039246)\n",
+      "('\"dangerously', 0.5920691192150116)\n",
```
Comment (piskvorky) [inline docs/notebooks/annoytutorial.ipynb]: This looks like bad preprocessing. Any reason not to simply use `simple_preprocess`?

File: docs/notebooks/annoytutorial.ipynb (hunk 1 above)
Comment (droudy) [inline docs/notebooks/annoytutorial.ipynb]: @piskvorky Should I round to a smaller decimal place or leave the exact figure out completely?

**Comment by piskvorky** [comment]
This doesn't look right -- I still see `"dangerously` in the notebook as a token, which should never happen with `simple_preprocess`.

**Pull request merged**